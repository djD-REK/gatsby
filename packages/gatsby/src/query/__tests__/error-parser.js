import errorParser from "../error-parser"

it.each([
  [
    `vague one`,
    `Encountered 1 error(s):
  - Unknown field 'fdnjkgd' on type 'SiteSiteMetadata'.`,
    `85907`,
  ],
  [`totally vague one`, `foo bar`, `85901`],
  [
    `Windows uppercase folders error`,
    `Error: RelayParser: Encountered duplicated definitions for one or more documents: each document must have a unique name. Duplicated documents:`,
    `85902`,
  ],
])(`%s`, (_name, message, expectedId) => {
  const structured = errorParser({
    message,
    filePath: `test.js`,
    location: { start: { line: 5, column: 10 } },
  })
  expect(structured).toMatchSnapshot()
  expect(structured.id).toEqual(expectedId)
})
