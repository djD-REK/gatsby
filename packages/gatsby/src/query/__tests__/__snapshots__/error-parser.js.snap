// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Windows uppercase folders error 1`] = `
Object {
  "context": Object {
    "sourceMessage": "Error: RelayParser: Encountered duplicated definitions for one or more documents: each document must have a unique name. Duplicated documents:

Warning: Windows users may be able to fix this error by renaming all directories to lowercase
in the file path and also by installing package node-win-env using npm install -g node-win-env",
  },
  "filePath": "test.js",
  "id": "85902",
  "location": Object {
    "start": Object {
      "column": 10,
      "line": 5,
    },
  },
}
`;

exports[`totally vague one 1`] = `
Object {
  "context": Object {
    "sourceMessage": "foo bar",
  },
  "filePath": "test.js",
  "id": "85901",
  "location": Object {
    "start": Object {
      "column": 10,
      "line": 5,
    },
  },
}
`;

exports[`vague one 1`] = `
Object {
  "context": Object {
    "message": "- Unknown field 'fdnjkgd' on type 'SiteSiteMetadata'.",
  },
  "filePath": "test.js",
  "id": "85907",
  "location": Object {
    "start": Object {
      "column": 10,
      "line": 5,
    },
  },
}
`;
